pkg/controller/container-runtime-config/helpers.go

```diff
@@ -496,15 +496,38 @@ func getValidBlockedRegistries(releaseImage string, imgSpec *apicfgv1.ImageSpec)
 	for i, reg := range imgSpec.RegistrySources.BlockedRegistries {
 		// if there is a match, return all the blocked registries except the one that matched and return an error as well
 		if reg == payloadReg {
-			blockedRegs = append(blockedRegs, imgSpec.RegistrySources.BlockedRegistries[i+1:]...)
-			return blockedRegs, fmt.Errorf("error adding %q to blocked registries, cannot block the registry being used by the payload", payloadReg)
+			// If the payload registry doesn't have mirror rules configured for it, then don't add it to the blocked registries list
```

Comment:
Mirrors can also be blocked; so this should probably verify that the mirrors that are defined are not blocked either.

Then the ugly questions:

And if there are multiple mirrors, and only a subset is blocked, I guess we allow that, as long as there is one unblocked mirror, arguing that a blocked mirror is the same as a blocked primary, and not _alone_ a reason to reject the whole configuration.

Alternatively, I could see an argument that any definition of a mirror pointing at a blocked repo is by definition invalid, and should be rejected. OTOH this controller can not “reject” configuration, it can only “ignore” it, and I’m not at all sure to what extent ignoring admin’s security policy to block a registry is acceptable.

Reply:
Added a check to go through the mirrors configured to check if a mirror has been configured for the payload reg, if so then verify that at least one of the mirrors configured is not blocked.